Automattic/newspack-ios
# Issues/59 order assets

Closes #59 

This PR wraps up the sorting logic for StoryAssets. The list of story assets can be filtered between two sort modes: asset type, and a user defined sort mode.  Support is also added for dragging items from the unsorted section into the sorted section, and reordering items in the sorted section.   There is some temporary UI to facilitate testing the different sort modes and the editing behavior.  Please note that the new edit button in the nav bar only applies when viewing the sorted/unsorted list.

### To Test:

Prep: Create a Story Folder, enter the folder and add some text notes. Go the Photos app and copy some photos. Use the Files app to paste these into the story folder you created.

Scenario 1: Type Mode
- View the assets list for your story folder and confirm that you see text and image assets in two different sections.  
- Confirm the first section has text notes, and the second section has images.

Scenario 2: Switching Modes + Rememberance
- Switch to Ordered mode and back a few times.  Confirm that the list shows the correct sections every time.
- Confirm that the sort mode you are viewing is remembered by returning to the list of Story Folders, then back to the list of Assets.  You should always be returned to the list mode you were previously viewing.  
- Confirm that the segment control always shows the correct mode selected.

Scenario 3: Ordered Mode - Deletion
- While viewing ordered mode tap the + button to add a new text note.
- Delete the note. Confirm that the correct note was deleted.

Scenario 4: Ordered Mode - Editing
- If you do not have both an ordered and unordered sections, tap a few unordered rows to make them ordered.
- While viewing ordered mode, tap the Edit button.  Confirm that the drag accessory icons appear.
- Confirm that you can drag an item from the unsorted list into the sorted list.
- Confirm that you can not change the order of unsorted items.
- Confirm that you can not drag an item from the sorted list into the unsorted list.
- Confirm that you can change the order of items in the sorted list.
- Return to the list of Story Folders, then back to the list of assets.  Confirm that the order is remembered.

Scenario 5: Blank canvas.
- View the list of assets for a story folder with no assets.
- Confirm nothing goes 💥. 


Note: You can tap a sorted row to make it unsorted, or to give it a sort order of 1.  This is a hack to make testing easier and will be removed in the future. :) 

### Examples:
#### The Type list:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-13 at 17 05 10](https://user-images.githubusercontent.com/1435271/87359491-855ebc80-c52d-11ea-83f3-e7cfd47127c4.png)

#### The Sorted list:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-13 at 17 05 30](https://user-images.githubusercontent.com/1435271/87359497-8859ad00-c52d-11ea-8a73-156853645e9a.png)

Hola @jleandroperez!  Game for a review?
